cmd/gateway-handlers.go

```diff
@@ -644,15 +644,6 @@ func (api gatewayAPIHandlers) PutBucketHandler(w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	// validating region here, because isValidLocationConstraint
-	// reads body which has been read already. So only validating
-	// region here.
-	serverRegion := serverConfig.GetRegion()
-	if serverRegion != location {
```

Comment:
We can use isValidLocation instead here. And return error properly. Since we need to honor if gateway region was set manually.